lib/src/request.dart

```diff
@@ -113,10 +113,10 @@ class Request extends BaseRequest {
           'content-type "application/x-www-form-urlencoded".');
     }
 
-    return Uri.splitQueryString(body, encoding: encoding);
+    return splitQueryString(body, encoding: encoding);
```

Comment:
Hmm, returning a `List<String>` value here will break backwards-compatibility, as will changing the return type to `Map<String, Object>`. I'm also not sure that returning a map that may have list or string keys is the right API in the abstract; it would make it easy for code written assuming strings to crash or, worse, behave unpredictably if the browser caused a list to be returned instead.

I think the ideal solution here would be to have a sort of multimap class that implements `Map<String, String>` but allows `[]=` to take a list and has a `.getMultiple()` method so users can opt in to getting multiple values. That sort of thing would also be really useful for headers. If you'd like to implement that (maybe in the `collection` package), it would be awesome; if you don't want to, maybe we should just make this feature write-only for now.